src/CreateEntityMappingView/CreateEntityMappingView.js

```diff
@@ -249,98 +312,192 @@ class CreateEntityMappingView extends ConnectionComponent {
+						floatingLabelText="Select Entity Type"
+						fullWidth={true}>
+						{CreateEntityMappingView.getMenuItems(initialData.entityTypes)}
+					</SelectField>
+				</Col>
+			</Row>
+		);
+	}
+
+	static renderAttributeFields(initialData) {
+		return initialData.attributeFields;
+	}
+
+	static renderButtons(submitCallback) {
+		return (
+			<div className={css(AppStyles.textAlignCenter)}>
+				<RaisedButton
+					label="Abort"
+					icon={<IconCancel/>}
+					className={css(AppStyles.marginAllSites)}
+					secondary={true}
+					onTouchTap={CreateEntityMappingView.abort}
+				/>
+				<RaisedButton
+					label="Save"
+					icon={<IconSave/>}
+					onTouchTap={submitCallback}
+					className={css(AppStyles.marginAllSites)}
+					primary={true}
+				/>
+			</div>
+		);
+	}
+
+	renderComponentBody(initialData, optionalActions, submitCallback) {
+		return (
+			<Container>
+				<div>
+					{optionalActions && optionalActions.rejected && <ErrorMessage message={optionalActions.reason} />}
+					{this.renderSelectTargetStatus()}
+					{this.renderSelectTypes(initialData)}
+					{CreateEntityMappingView.renderAttributeFields(initialData)}
+					{CreateEntityMappingView.renderButtons(submitCallback)}
+				</div>
+			</Container>
+		);
+	}
+
+	fetchInitialData() {
 		const allFetches = PromiseState.all([this.props.eventTypes, this.props.entityTypeHierarchy]);
 		const eventTypes = this.props.eventTypes.value;
-		const optionalActions = this.props.createEntityMappingResponse;
-		if(optionalActions && optionalActions.fulfilled) {
-			window.history.back();
-			return null;
-		}
-		const entityTypes = this.transformHierarchy(this.props.entityTypeHierarchy.value);
+		const entityTypes = CreateEntityMappingView.transformHierarchy(this.props.entityTypeHierarchy.value);
 		const connectionIncomplete = super.render(allFetches);
 		if (connectionIncomplete) {
-			return connectionIncomplete;
+			return false;
 		}
 		let attributeFields = "";
-		if (this.props.eventTypeAttributes && this.props.entityTypeAttributes 
+		if (this.props.eventTypeAttributes && this.props.entityTypeAttributes
 			&& this.props.eventTypeAttributes.fulfilled && this.props.entityTypeAttributes.fulfilled) {
-			attributeFields = this.loadAttributes();
+			attributeFields = this.loadAttributeInputFields();
+		}
+
+		return {
+			eventTypes: eventTypes,
+			entityTypes: entityTypes,
+			attributeFields: attributeFields
+		};
+	}
+
+	renderEditView() {
+		const optionalActions = this.props.updateEntityMappingResponse;
+		CreateEntityMappingView.handleOptionalActionsSuccess(optionalActions);
+		const initialData = this.fetchInitialData();
+		if (!initialData) {
+			return <LoadingAnimation/>;
+		}
+		return (
+			<div ref={() => {this.oldValuesShouldBeLoaded = true;}}>
+				<Header title={"Edit Entity Mapping"}/>
+				<div className={AppStyles.elementMarginTop}>
+					{this.renderComponentBody(initialData, optionalActions, this.submitUpdatedMapping)}
```

Comment:
ComponentBody: Insufficient naming
Component refers to entities of react framework ...